templates/firstboot/mellanox_fw_update.yaml

```diff
@@ -138,13 +147,14 @@ resources:
                         return self._devs
 
                     devs = []
-                    cmd = ['lspci', '-d', '15b3:']
+                    cmd = ['lspci', '-D','-d', '15b3:']
                     out = run_command(*cmd)
                     for line in out.split('\n'):
                         if not line:
                             continue
                         dev = line.split()[0]
-                        if dev.endswith('.0'):
+                        if dev.endswith('.0') and (not self._dev_white_list or
+                                dev in self._dev_white_list):
```

Comment:
we should log a warning all the pci that are in the whitelist and not found

Reply:
Also if there are pci that ends with not zero in the whitelist we should log a warning

Reply:
> we should log a warning all the pci that are in the whitelist and not found

Done

Reply:
> Also if there are pci that ends with not zero in the whitelist we should log a warning

Done

Reply:
you are breaking per PF configuration here